back/app/controllers/web_api/v1/projects_controller.rb

```diff
@@ -52,6 +52,74 @@ def index
     )
   end
 
+  # For use with 'Open to participation' homepage widget.
+  # Returns all published or archived projects that are visible to user
+  # and in an active participatory phase (where user can do something).
+  # Ordered by the end date of the current phase, soonest first (nulls last).
+  def index_projects_with_active_participatory_phase
```

Comment:
This is quite complex in a controller. It might be easier to isolate and test and be more resuable if the logic was in a service somewhere eg (Psuedo code) `ProjectsFinder.projects_open_to_participation`